sky/packages/sky/lib/src/fn3/focus.dart

```diff
@@ -137,8 +140,9 @@ class FocusState extends State<Focus> {
   void _scopeRemoved(GlobalKey key) {
     assert(_focusedScope == key);
     _currentlyRegisteredScopeRemovalListenerKey = null;
-    setState(() {
-      _focusedScope = null;
+    _focusedScope = null;
```

Comment:
But you update the _focusedScope state, so if it happens to build after doing so, the effect will be applied this round.